src/App.tsx

```diff
@@ -15,7 +15,7 @@ type FilterStatus = 'all' | 'active' | 'completed';
 interface DraggableItemProps {
   id: string;
   index: number;
-  moveItem: (dragIndex: number, hoverIndex: number) => void;
+  moveItem: (dragIndex: number, hoverIndex: number, fromGroup: string, toGroup: string) => void;
   children: React.ReactNode;
```

Comment:
I'm getting an error saying React is not defined.

Reply:
It's working when I drag from 'today' to 'tomorrow' but the app breaks if I try to drag into 'upcoming'. I get this error in the console:
```
App.tsx:123 Uncaught TypeError: Cannot set properties of undefined (setting 'dueDate')
    at App.tsx:123:21
    at basicStateReducer (react-dom_client.js?v=004121f9:11703:49)
    at updateReducer (react-dom_client.js?v=004121f9:11794:30)
    at updateState (react-dom_client.js?v=004121f9:12021:18)
    at Object.useState (react-dom_client.js?v=004121f9:12753:24)
    at useState (chunk-DRWLMN53.js?v=004121f9:1066:29)
    at App (App.tsx:78:29)
    at renderWithHooks (react-dom_client.js?v=004121f9:11548:26)
    at updateFunctionComponent (react-dom_client.js?v=004121f9:14582:28)
    at beginWork (react-dom_client.js?v=004121f9:15924:22)
(anonymous) @ App.tsx:123
basicStateReducer @ react-dom_client.js?v=004121f9:11703
updateReducer @ react-dom_client.js?v=004121f9:11794
updateState @ react-dom_client.js?v=004121f9:12021
useState @ react-dom_client.js?v=004121f9:12753
useState @ chunk-DRWLMN53.js?v=004121f9:1066
App @ App.tsx:78
renderWithHooks @ react-dom_client.js?v=004121f9:11548
updateFunctionComponent @ react-dom_client.js?v=004121f9:14582
beginWork @ react-dom_client.js?v=004121f9:15924
callCallback2 @ react-dom_client.js?v=004121f9:3674
invokeGuardedCallbackDev @ react-dom_client.js?v=004121f9:3699
invokeGuardedCallback @ react-dom_client.js?v=004121f9:3733
beginWork$1 @ react-dom_client.js?v=004121f9:19765
performUnitOfWork @ react-dom_client.js?v=004121f9:19198
workLoopSync @ react-dom_client.js?v=004121f9:19137
renderRootSync @ react-dom_client.js?v=004121f9:19116
performConcurrentWorkOnRoot @ react-dom_client.js?v=004121f9:18678
workLoop @ react-dom_client.js?v=004121f9:197
flushWork @ react-dom_client.js?v=004121f9:176
performWorkUntilDeadline @ react-dom_client.js?v=004121f9:384
Show 19 more frames
Show less
App.tsx:123 Uncaught TypeError: Cannot set properties of undefined (setting 'dueDate')
    at App.tsx:123:21
    at basicStateReducer (react-dom_client.js?v=004121f9:11703:49)
    at updateReducer (react-dom_client.js?v=004121f9:11794:30)
    at updateState (react-dom_client.js?v=004121f9:12021:18)
    at Object.useState (react-dom_client.js?v=004121f9:12753:24)
    at useState (chunk-DRWLMN53.js?v=004121f9:1066:29)
    at App (App.tsx:78:29)
    at renderWithHooks (react-dom_client.js?v=004121f9:11548:26)
    at updateFunctionComponent (react-dom_client.js?v=004121f9:14582:28)
    at beginWork (react-dom_client.js?v=004121f9:15924:22)
(anonymous) @ App.tsx:123
basicStateReducer @ react-dom_client.js?v=004121f9:11703
updateReducer @ react-dom_client.js?v=004121f9:11794
updateState @ react-dom_client.js?v=004121f9:12021
useState @ react-dom_client.js?v=004121f9:12753
useState @ chunk-DRWLMN53.js?v=004121f9:1066
App @ App.tsx:78
renderWithHooks @ react-dom_client.js?v=004121f9:11548
updateFunctionComponent @ react-dom_client.js?v=004121f9:14582
beginWork @ react-dom_client.js?v=004121f9:15924
callCallback2 @ react-dom_client.js?v=004121f9:3674
invokeGuardedCallbackDev @ react-dom_client.js?v=004121f9:3699
invokeGuardedCallback @ react-dom_client.js?v=004121f9:3733
beginWork$1 @ react-dom_client.js?v=004121f9:19765
performUnitOfWork @ react-dom_client.js?v=004121f9:19198
workLoopSync @ react-dom_client.js?v=004121f9:19137
renderRootSync @ react-dom_client.js?v=004121f9:19116
recoverFromConcurrentError @ react-dom_client.js?v=004121f9:18736
performConcurrentWorkOnRoot @ react-dom_client.js?v=004121f9:18684
workLoop @ react-dom_client.js?v=004121f9:197
flushWork @ react-dom_client.js?v=004121f9:176
performWorkUntilDeadline @ react-dom_client.js?v=004121f9:384
Show 20 more frames
Show less
console.js:288 The above error occurred in the <App> component:

    at App (http://localhost:5174/src/App.tsx:81:29)

Consider adding an error boundary to your tree to customize error handling behavior.
Visit https://reactjs.org/link/error-boundaries to learn more about error boundaries.
overrideMethod @ console.js:288
logCapturedError @ react-dom_client.js?v=004121f9:14032
update.callback @ react-dom_client.js?v=004121f9:14052
callCallback @ react-dom_client.js?v=004121f9:11248
commitUpdateQueue @ react-dom_client.js?v=004121f9:11265
commitLayoutEffectOnFiber @ react-dom_client.js?v=004121f9:17093
commitLayoutMountEffects_complete @ react-dom_client.js?v=004121f9:17980
commitLayoutEffects_begin @ react-dom_client.js?v=004121f9:17969
commitLayoutEffects @ react-dom_client.js?v=004121f9:17920
commitRootImpl @ react-dom_client.js?v=004121f9:19353
commitRoot @ react-dom_client.js?v=004121f9:19277
finishConcurrentRender @ react-dom_client.js?v=004121f9:18760
performConcurrentWorkOnRoot @ react-dom_client.js?v=004121f9:18718
workLoop @ react-dom_client.js?v=004121f9:197
flushWork @ react-dom_client.js?v=004121f9:176
performWorkUntilDeadline @ react-dom_client.js?v=004121f9:384
Show 16 more frames
Show less
react-dom_client.js?v=004121f9:19413 Uncaught TypeError: Cannot set properties of undefined (setting 'dueDate')
    at App.tsx:123:21
    at basicStateReducer (react-dom_client.js?v=004121f9:11703:49)
    at updateReducer (react-dom_client.js?v=004121f9:11794:30)
    at updateState (react-dom_client.js?v=004121f9:12021:18)
```

Reply:
Now I'm getting this error:
```
Uncaught ReferenceError: Cannot access 'isDueOrOverdue' before initialization
```

Reply:
Hmm, now there's some weird behaviour where if I have one item in each group, and drag a 'today' item, hover over the 'tomorrow' group while dragging, and then hover over the 'upcoming' group while dragging, all 3 items end up in the 'upcoming' group, even though I was only dragging the 'today' item.

Reply:
Ok now it is working ok, but there's no visual feedback when we are dragging an item and hovering over a group. Can we show a dotted box where the item will be dropped, while we are dragging? Keep in mind this earlier issue:
Hmm, now there's some weird behaviour where if I have one item in each group, and drag a 'today' item, hover over the 'tomorrow' group while dragging, and then hover over the 'upcoming' group while dragging, all 3 items end up in the 'upcoming' group, even though I was only dragging the 'today' item.